runtime/gc_modron_standard/StandardAccessBarrier.cpp

```diff
@@ -92,6 +105,135 @@ MM_StandardAccessBarrier::tearDown(MM_EnvironmentBase *env)
 	MM_ObjectAccessBarrier::tearDown(env);
 }
 
+/**
+ * Unmarked, heap reference, about to be deleted (or overwritten), while marking
+ * is in progress is to be remembered for later marking and scanning.
+ */
+void
+MM_StandardAccessBarrier::rememberObject(MM_EnvironmentBase *env, J9Object *object)
+{
+	//TODO Evgenia SATBBarrier
+	//if (_markingScheme->markObject(env, object, true)) {
+		rememberObjectImpl(env, object);
+	//}
+}
+
+/**
+ * Unmarked, heap reference, about to be deleted (or overwritten), while marking
+ * is in progress is to be remembered for later marking and scanning.
+ * This method is called by MM_RealtimeAccessBarrier::rememberObject()
+ */
+void
+MM_StandardAccessBarrier::rememberObjectImpl(MM_EnvironmentBase *env, J9Object* object)
+{
+	J9VMThread *vmThread = (J9VMThread *)env->getLanguageVMThread();
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(vmThread->javaVM);
+
+	extensions->sATBBarrierRememberedSet->storeInFragment(env, &vmThread->sATBBarrierRememberedSetFragment, (UDATA *)object);
+}
+
+
+bool
+MM_StandardAccessBarrier::preObjectStoreInternal(J9VMThread *vmThread, J9Object *destObject, fj9object_t *destAddress, J9Object *value, bool isVolatile)
+{
+	J9JavaVM *javaVM = vmThread->javaVM;
+	if (javaVM->gcWriteBarrierType != gc_modron_wrtbar_realtime) {
+		return true;
+	}
+	MM_EnvironmentBase* env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+
+	if (isBarrierActive(env)) {
```

Comment:
these two isBarrierActive method names should be more specific (refer to SATB)

Reply:
renamed the helper to isSATBBarrierActive.